src/features/soft_navigations/aggregate/ajax-node.js

```diff
@@ -0,0 +1,57 @@
+import { addCustomAttributes, getAddStringContext, nullable, numeric } from '../../../common/serialize/bel-serializer'
+import { NODE_TYPE } from '../constants'
+import { BelNode } from './bel-node'
+
+export class AjaxNode extends BelNode {
+  constructor (agentIdentifier, ajaxEvent) {
+    super(agentIdentifier)
+    this.belType = NODE_TYPE.AJAX
+    this.method = ajaxEvent.method
+    this.status = ajaxEvent.status
+    this.domain = ajaxEvent.domain
+    this.path = ajaxEvent.path
+    this.txSize = ajaxEvent.requestSize
+    this.rxSize = ajaxEvent.responseSize
+    this.requestedWith = ajaxEvent.type === 'fetch' ? 1 : '' // 'xhr' and 'beacon' types get the empty string
+    this.spanId = ajaxEvent.spanId
+    this.traceId = ajaxEvent.traceId
+    this.spanTimestamp = ajaxEvent.spanTimestamp
+    this.gql = ajaxEvent.gql
+
+    this.start = ajaxEvent.startTime
+    this.end = ajaxEvent.endTime
+  }
+
+  serialize (parentStartTimestamp) {
+    const addString = getAddStringContext(this.agentIdentifier)
+    const nodeList = []
+
+    // IMPORTANT: The order in which addString is called matters and correlates to the order in which string shows up in the harvest payload. Do not re-order the following code.
+    const fields = [
+      numeric(this.belType),
+      0, // this will be overwritten below with number of attached nodes
+      numeric(this.start - parentStartTimestamp), // start relative to parent start (if part of first node in payload) or first parent start
+      numeric(this.end - this.start), // end is relative to start
+      numeric(this.callbackEnd),
+      numeric(this.callbackDuration),
+      addString(this.method),
+      numeric(this.status),
+      addString(this.domain),
+      addString(this.path),
+      numeric(this.txSize),
+      numeric(this.rxSize),
+      this.requestedWith,
+      addString(this.nodeId),
+      nullable(this.spanId, addString, true) + nullable(this.traceId, addString, true) + nullable(this.spanTimestamp, numeric)
+    ]
+    let allAttachedNodes = []
+    if (typeof this.gql === 'object') allAttachedNodes = addCustomAttributes(this.gql, addString)
```

Comment:
Don't ajax nodes under old spa get other custom attributes besides just GQL?  Or is that handled somewhere else?

Reply:
In `serializeInteraction` for serializer.js, for type 2 (ajax) nodes, only that is tacked onto `children` and no other attributes. You may be thinking of bel.7;2 payloads handled directly by the ajax feature itself, in which case it prepends `jsAttributes` as well.

Reply:
does it inherit the custom attributes for ajax children from the parent browser ixn then?

Reply:
Which is to say that the [staging app on NR1](https://one.newrelic.com/nr1-core/ajax-requests/home/Mjk0MjgxMXxCUk9XU0VSfEFQUExJQ0FUSU9OfDQwNzUwNDQ3NQ?duration=1800000&filters=%28domain%20IN%20%28%27BROWSER%27%2C%20%27BROWSER%27%29%20AND%20type%20IN%20%28%27APPLICATION%27%2C%20%27MICRO_FRONTEND%27%29%29&state=bfdad770-a66b-849a-b4e7-830133a13da1) as an example may be overcounting.

Reply:
Thanks, please note this as a follow up item to investigate further in the GQL CDD